cmd/runner/main.go

```diff
@@ -1,56 +1,76 @@
 package main
 
 import (
-	"context"
+	"errors"
 	"flag"
 	"fmt"
+	"log"
 	"os"
 	"time"
 
+	"github.com/benchttp/runner/config"
+	"github.com/benchttp/runner/configparser"
 	"github.com/benchttp/runner/request"
 )
 
-const (
-	DefaultConcurrency = 1
-	DefaultRequests    = 0 // Use duration as exit condition if omitted.
-	DefaultDuration    = 60 * time.Second
-	DefaultTimeout     = 10 * time.Second
-)
-
 var (
-	url         string
-	concurrency int           // Number of connections to run concurrently.
-	requests    int           // Number of requests to run, use duration as exit condition if omitted.
-	duration    time.Duration // Duration of test, in seconds.
-	timeout     time.Duration // Timeout for each http request, in seconds.
+	configFile    string
+	url           string
+	concurrency   int           // Number of connections to run concurrently
+	requests      int           // Number of requests to run, use duration as exit condition if omitted.
+	timeout       time.Duration // Timeout for each http request
+	globalTimeout time.Duration // Duration of test
 )
 
-func parseArgs() {
-	url = os.Args[len(os.Args)-1]
+var defaultConfigFiles = []string{
+	"./.benchttp.yml",
+	"./.benchttp.yaml",
+	"./.benchttp.json",
+}
 
-	flag.IntVar(&concurrency, "c", DefaultConcurrency, "Number of connections to run concurrently")
-	flag.IntVar(&requests, "r", DefaultRequests, "Number of requests to run, use duration as exit condition if omitted")
-	flag.DurationVar(&duration, "d", DefaultDuration, "Duration of test, in seconds")
-	flag.DurationVar(&timeout, "t", DefaultTimeout, "Timeout for each http request, in seconds")
+func parseArgs() {
+	flag.StringVar(&configFile, "configFile", findFile(defaultConfigFiles), "Config file path")
+	flag.StringVar(&url, "url", "", "Target URL to request")
+	flag.IntVar(&concurrency, "concurrency", 0, "Number of connections to run concurrently")
```

Comment:
Shouldn't we rather use 1 for default concurrency? Since no concurrency = 1 worker at a time.

Reply:
5d4fd1d59e78ef3d6a5bedef26e84be84bcccfcf
> Rework config defaulting
>
> - use zero values for cli defaults (so they don't override when not set)
> - use config.Default() for default config

We need to distinguish the CLI default values from the config defaults. The reason is we override the options from config file with the CLI options _when they are set_. If we use `1` as CLI default, it will be considered set to `1` by default and the config file value will always be ignored.